src/main/java/org/truffleruby/core/array/ArrayBuilderNode.java

```diff
@@ -25,6 +25,37 @@
 
 public abstract class ArrayBuilderNode extends RubyBaseNode {
 
+    public static class ArrayBuilderProxyNode extends RubyBaseNode {
```

Comment:
Why not just `extends ArrayBuilderNode` here and keep the same interface, so caller code can remain identical?